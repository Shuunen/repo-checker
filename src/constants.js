import { join } from 'path'

export const home = process.env.HOME
export const dataFileName = 'repo-checker.config.js'
export const repoCheckerPath = join(__dirname, '..')
export const templatePath = join(repoCheckerPath, 'templates')
export const dataFileHomePath = join(home, dataFileName)

export const dataDefaults = {
  auto_merge: true,
  ban_sass: true,
  dev_deps_only: true,
  license: 'GPL-3.0',
  max_size_ko: 10,
  user_mail: 'romain.racamier@gmail.com',
  user_name: 'Romain Racamier-Lafon',
}
