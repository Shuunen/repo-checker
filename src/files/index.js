const CheckConfigs = require('./configs')
const CheckReadme = require('./readme')
const CheckPackage = require('./package')
const CheckLicense = require('./license')
const CheckTravis = require('./travis')
const CheckRenovate = require('./renovate')
const CheckEditorConfig = require('./editor-config')
const CheckNvmRc = require('./nvmrc')

module.exports = [CheckConfigs, CheckNvmRc, CheckReadme, CheckPackage, CheckLicense, CheckTravis, CheckRenovate, CheckEditorConfig]
