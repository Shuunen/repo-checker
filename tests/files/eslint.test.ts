import { expect, it } from 'vitest'
import { ProjectData } from '../../src/constants'
import { EsLintFile } from '../../src/files'
import { log } from '../../src/logger'
import { cleanInstanceForSnap, promiseFalse, promiseTrue, promiseVoid, tsProjectFolder, vueProjectFolder } from '../utils'

it('eslint A config missing file', async () => {
  log.disable()
  const instance = new EsLintFile()
  instance.fileExists = promiseFalse
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('eslint B config file empty', async () => {
  log.disable()
  const instance = new EsLintFile()
  instance.fileExists = promiseTrue
  instance.inspectFile = promiseVoid
  instance.fileContent = ''
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('eslint C config file empty for ts project', async () => {
  log.disable()
  const instance = new EsLintFile('', new ProjectData({ isUsingTypescript: true }))
  instance.fileExists = promiseTrue
  instance.inspectFile = promiseVoid
  instance.fileContent = ''
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('eslint D config file empty for vue ts project', async () => {
  log.disable()
  const instance = new EsLintFile('', new ProjectData({ isUsingVue: true, isUsingTypescript: true }))
  instance.fileExists = promiseTrue
  instance.inspectFile = promiseVoid
  instance.fileContent = ''
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('eslint E config file empty for tailwind project', async () => {
  log.disable()
  const instance = new EsLintFile('', new ProjectData({ isUsingTailwind: true }))
  instance.fileExists = promiseTrue
  instance.inspectFile = promiseVoid
  instance.fileContent = ''
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('eslint F config partial file for js project', async () => {
  log.disable()
  const instance = new EsLintFile()
  instance.fileExists = promiseTrue
  instance.inspectFile = promiseVoid
  instance.fileContent = `{
    "extends": [
      "eslint:recommended",
      "plugin:unicorn/recommended"
    ],
    "plugins": [
      "unicorn"
    ],
    "rules": {}
  }`
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('eslint G config partial file for ts project', async () => {
  log.disable()
  const instance = new EsLintFile('', new ProjectData({ isUsingTypescript: true }))
  instance.fileExists = promiseTrue
  instance.inspectFile = promiseVoid
  instance.fileContent = `{
    "extends": [
      "eslint:recommended",
      "plugin:@typescript-eslint/recommended",
      "plugin:unicorn/recommended"
    ],
    "rules": {}
  }`
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('eslint H config partial file for vue ts project', async () => {
  log.disable()
  const instance = new EsLintFile('', new ProjectData({ isUsingVue: true, isUsingTypescript: true }))
  instance.fileExists = promiseTrue
  instance.inspectFile = promiseVoid
  instance.fileContent = `{
    "extends": [
      "eslint:recommended",
      "plugin:@typescript-eslint/recommended",
      "plugin:unicorn/recommended",
      "plugin:vue/vue3-recommended"
    ],
    "parser": "vue-eslint-parser",
    "parserOptions": {
      "parser": "@typescript-eslint/parser",
      "ecmaVersion": 2021,
      "sourceType": "module"
    },
    "plugins": [
      "@typescript-eslint",
      "unicorn"
    ],
    "rules": {}
  }`
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('eslint I up to date config file for vue ts project', async () => {
  log.disable()
  const instance = new EsLintFile('', new ProjectData({ isUsingVue: true, isUsingTypescript: true }))
  instance.fileExists = promiseTrue
  instance.inspectFile = promiseVoid
  instance.fileContent = `{
    "extends": [
      "plugin:vue/vue3-recommended",
      "eslint:recommended",
      "@vue/typescript/recommended",
      "plugin:unicorn/recommended"
    ],
    "parserOptions": {
      "ecmaVersion": 2021
    },
    "plugins": [
      "html",
      "unicorn"
    ],
  }`
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('eslint J config file with no rules', async () => {
  log.disable()
  const instance = new EsLintFile(vueProjectFolder, new ProjectData({}))
  instance.fileExists = promiseTrue
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})

it('eslint K config file with just rules (no override)', async () => {
  log.disable()
  const instance = new EsLintFile(tsProjectFolder, new ProjectData({}))
  instance.fileExists = promiseTrue
  await instance.start()
  await instance.end()
  expect(cleanInstanceForSnap(instance)).toMatchSnapshot()
})
