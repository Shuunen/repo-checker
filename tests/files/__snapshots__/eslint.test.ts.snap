// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`eslint A config missing file 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [],
  "fileContent": "",
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "has-no-xo-config-js-file",
  ],
  "warnings": [],
}
`;

exports[`eslint B config file empty 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [
    "has-no-xo-config-js-file",
  ],
  "fileContent": "",
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "has-no-promise-plugin-require-eslint-7",
  ],
  "warnings": [
    "could-have-hardcore-rules-extend-hardcore",
    "could-have-eslint-recommended-rules-extend-eslint-recommended",
    "could-have-unicorn-rules-extend-plugin-unicorn-all",
    "current-eslintrc-json-has-only-14-of-the-53-custom-rules-in-repo-checker-eslintrc-json",
  ],
}
`;

exports[`eslint C config file empty for ts project 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": true,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [
    "has-no-xo-config-js-file",
  ],
  "fileContent": "",
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "has-no-promise-plugin-require-eslint-7",
    "has-a-disabled-explicit-function-return-type",
  ],
  "warnings": [
    "could-have-hardcore-rules-extend-hardcore",
    "could-have-eslint-recommended-rules-extend-eslint-recommended",
    "could-have-unicorn-rules-extend-plugin-unicorn-all",
    "current-eslintrc-json-has-only-0-of-the-53-custom-rules-in-repo-checker-eslintrc-json",
    "could-have-hardcore-typescript-rules-extend-hardcore-ts",
  ],
}
`;

exports[`eslint D config file empty for vue ts project 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": true,
    "isUsingVue": true,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [
    "has-no-xo-config-js-file",
    "does-not-have-hardcore-vue-rules-extend-hardcore-vue",
  ],
  "fileContent": "",
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "has-no-promise-plugin-require-eslint-7",
    "has-a-disabled-explicit-function-return-type",
  ],
  "warnings": [
    "could-have-hardcore-rules-extend-hardcore",
    "could-have-eslint-recommended-rules-extend-eslint-recommended",
    "could-have-unicorn-rules-extend-plugin-unicorn-all",
    "current-eslintrc-json-has-only-0-of-the-60-custom-rules-in-repo-checker-eslintrc-json",
    "could-have-hardcore-typescript-rules-extend-hardcore-ts",
  ],
}
`;

exports[`eslint E config file empty for tailwind project 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": true,
    "isUsingTypescript": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [
    "has-no-xo-config-js-file",
    "does-not-have-tailwind-rules-extend-plugin-tailwindcss-recommended",
  ],
  "fileContent": "",
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "has-no-promise-plugin-require-eslint-7",
  ],
  "warnings": [
    "could-have-hardcore-rules-extend-hardcore",
    "could-have-eslint-recommended-rules-extend-eslint-recommended",
    "could-have-unicorn-rules-extend-plugin-unicorn-all",
    "current-eslintrc-json-has-only-14-of-the-53-custom-rules-in-repo-checker-eslintrc-json",
  ],
}
`;

exports[`eslint F config partial file for js project 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [
    "has-no-xo-config-js-file",
  ],
  "fileContent": "{
    \\"extends\\": [
      \\"eslint:recommended\\",
      \\"plugin:unicorn/recommended\\"
    ],
    \\"plugins\\": [
      \\"unicorn\\"
    ],
    \\"rules\\": {}
  }",
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "has-eslint-recommended-rules-extend",
    "has-no-promise-plugin-require-eslint-7",
  ],
  "warnings": [
    "could-have-hardcore-rules-extend-hardcore",
    "could-have-unicorn-rules-extend-plugin-unicorn-all",
    "current-eslintrc-json-has-only-14-of-the-53-custom-rules-in-repo-checker-eslintrc-json",
  ],
}
`;

exports[`eslint G config partial file for ts project 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": true,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [
    "has-no-xo-config-js-file",
  ],
  "fileContent": "{
    \\"extends\\": [
      \\"eslint:recommended\\",
      \\"plugin:@typescript-eslint/recommended\\",
      \\"plugin:unicorn/recommended\\"
    ],
    \\"rules\\": {}
  }",
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "has-eslint-recommended-rules-extend",
    "has-no-promise-plugin-require-eslint-7",
    "has-a-disabled-explicit-function-return-type",
  ],
  "warnings": [
    "could-have-hardcore-rules-extend-hardcore",
    "could-have-unicorn-rules-extend-plugin-unicorn-all",
    "current-eslintrc-json-has-only-0-of-the-53-custom-rules-in-repo-checker-eslintrc-json",
    "could-have-hardcore-typescript-rules-extend-hardcore-ts",
  ],
}
`;

exports[`eslint H config partial file for vue ts project 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": true,
    "isUsingVue": true,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [
    "has-no-xo-config-js-file",
    "does-not-have-hardcore-vue-rules-extend-hardcore-vue",
  ],
  "fileContent": "{
    \\"extends\\": [
      \\"eslint:recommended\\",
      \\"plugin:@typescript-eslint/recommended\\",
      \\"plugin:unicorn/recommended\\",
      \\"plugin:vue/vue3-recommended\\"
    ],
    \\"parser\\": \\"vue-eslint-parser\\",
    \\"parserOptions\\": {
      \\"parser\\": \\"@typescript-eslint/parser\\",
      \\"ecmaVersion\\": 2021,
      \\"sourceType\\": \\"module\\"
    },
    \\"plugins\\": [
      \\"@typescript-eslint\\",
      \\"unicorn\\"
    ],
    \\"rules\\": {}
  }",
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "has-eslint-recommended-rules-extend",
    "has-no-promise-plugin-require-eslint-7",
    "has-a-disabled-explicit-function-return-type",
  ],
  "warnings": [
    "could-have-hardcore-rules-extend-hardcore",
    "could-have-unicorn-rules-extend-plugin-unicorn-all",
    "current-eslintrc-json-has-only-0-of-the-60-custom-rules-in-repo-checker-eslintrc-json",
    "could-have-hardcore-typescript-rules-extend-hardcore-ts",
  ],
}
`;

exports[`eslint I up to date config file for vue ts project 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": true,
    "isUsingVue": true,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [
    "has-no-xo-config-js-file",
    "does-not-have-hardcore-vue-rules-extend-hardcore-vue",
  ],
  "fileContent": "{
    \\"extends\\": [
      \\"plugin:vue/vue3-recommended\\",
      \\"eslint:recommended\\",
      \\"@vue/typescript/recommended\\",
      \\"plugin:unicorn/recommended\\"
    ],
    \\"parserOptions\\": {
      \\"ecmaVersion\\": 2021
    },
    \\"plugins\\": [
      \\"html\\",
      \\"unicorn\\"
    ],
  }",
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "has-eslint-recommended-rules-extend",
    "has-no-promise-plugin-require-eslint-7",
    "has-a-disabled-explicit-function-return-type",
  ],
  "warnings": [
    "could-have-hardcore-rules-extend-hardcore",
    "could-have-unicorn-rules-extend-plugin-unicorn-all",
    "could-have-hardcore-typescript-rules-extend-hardcore-ts",
  ],
}
`;

exports[`eslint J config file with no rules 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [
    "has-no-xo-config-js-file",
  ],
  "fileContent": "{
  \\"env\\": {
    \\"browser\\": false
  },
  \\"settings\\": {
    \\"repo-faker\\": {
      \\"hardcore/ts\\": \\"\\"
    }
  }
}
",
  "fileName": ".eslintrc.json",
  "originalFileContent": "{
  \\"env\\": {
    \\"browser\\": false
  },
  \\"settings\\": {
    \\"repo-faker\\": {
      \\"hardcore/ts\\": \\"\\"
    }
  }
}
",
  "passed": [
    "eslintrc-json-has-hardcore-rules-extend",
    "eslintrc-json-has-no-promise-plugin-require-eslint-7",
  ],
  "warnings": [
    "eslintrc-json-could-have-unicorn-rules-extend-plugin-unicorn-all",
    "eslintrc-json-current-eslintrc-json-has-only-14-of-the-53-custom-rules-in-repo-checker-eslintrc-json",
  ],
}
`;

exports[`eslint K config file with just rules (no override) 1`] = `
{
  "canFix": false,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [
    "has-no-xo-config-js-file",
  ],
  "fileContent": "{
  \\"env\\": {
    \\"browser\\": false
  },
  \\"rules\\": {
    \\"no-console\\": \\"off\\"
  },
  \\"settings\\": {
    \\"repo-faker\\": {
      \\"hardcore/ts\\": \\"\\"
    }
  }
}
",
  "fileName": ".eslintrc.json",
  "originalFileContent": "{
  \\"env\\": {
    \\"browser\\": false
  },
  \\"rules\\": {
    \\"no-console\\": \\"off\\"
  },
  \\"settings\\": {
    \\"repo-faker\\": {
      \\"hardcore/ts\\": \\"\\"
    }
  }
}
",
  "passed": [
    "eslintrc-json-has-hardcore-rules-extend",
    "eslintrc-json-has-no-promise-plugin-require-eslint-7",
  ],
  "warnings": [
    "eslintrc-json-could-have-unicorn-rules-extend-plugin-unicorn-all",
    "eslintrc-json-current-eslintrc-json-has-only-15-of-the-53-custom-rules-in-repo-checker-eslintrc-json",
  ],
}
`;

exports[`eslint L fix file content 1`] = `
{
  "canFix": true,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [
    "has-no-xo-config-js-file",
  ],
  "fileContent": "{
  \\"extends\\": [
    \\"eslint:recommended\\",
    \\"plugin:@typescript-eslint/recommended\\",
    \\"plugin:unicorn/recommended\\"
  ],
  \\"rules\\": {}
}",
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "has-eslint-recommended-rules-extend",
    "has-no-promise-plugin-require-eslint-7",
  ],
  "warnings": [
    "could-have-hardcore-rules-extend-hardcore",
    "could-have-unicorn-rules-extend-plugin-unicorn-all",
  ],
}
`;

exports[`eslint M fix rules in overrides 1`] = `
{
  "canFix": true,
  "canForce": false,
  "data": {
    "canAutoMergeDeps": true,
    "hasMainBranch": false,
    "hasTaskPrefix": false,
    "isModule": false,
    "isPublishedPackage": false,
    "isQuiet": false,
    "isUsingC8": false,
    "isUsingDependencyCruiser": false,
    "isUsingEslint": false,
    "isUsingNyc": false,
    "isUsingShuutils": false,
    "isUsingTailwind": false,
    "isUsingTypescript": false,
    "isUsingVue": false,
    "isWebPublished": false,
    "license": "GPL-3.0",
    "maxSizeKo": 1,
    "packageName": "",
    "repoId": "a-great-repo",
    "shouldAvoidSass": true,
    "userId": "Shuunen",
    "userIdLowercase": "shuunen",
    "userMail": "romain.racamier@gmail.com",
    "userName": "Romain Racamier-Lafon",
    "webUrl": "https://my-website.com",
    "willGenerateReport": true,
  },
  "failed": [
    "has-no-xo-config-js-file",
  ],
  "fileContent": "{
  \\"extends\\": [
    \\"eslint:recommended\\",
    \\"plugin:@typescript-eslint/recommended\\",
    \\"plugin:unicorn/recommended\\"
  ],
  \\"overrides\\": [
    {
      \\"files\\": [
        \\"*.ts\\"
      ],
      \\"rules\\": {
        \\"@typescript-eslint/brace-style\\": [
          \\"error\\",
          \\"1tbs\\",
          {
            \\"allowSingleLine\\": true
          }
        ],
        \\"@typescript-eslint/comma-dangle\\": [
          \\"error\\",
          \\"always-multiline\\"
        ],
        \\"@typescript-eslint/consistent-indexed-object-style\\": \\"off\\",
        \\"@typescript-eslint/consistent-type-definitions\\": \\"off\\",
        \\"@typescript-eslint/explicit-function-return-type\\": \\"off\\",
        \\"@typescript-eslint/indent\\": [
          \\"error\\",
          2
        ],
        \\"@typescript-eslint/lines-around-comment\\": \\"off\\",
        \\"@typescript-eslint/member-delimiter-style\\": [
          \\"error\\",
          {
            \\"multiline\\": {
              \\"delimiter\\": \\"none\\"
            }
          }
        ],
        \\"@typescript-eslint/no-type-alias\\": \\"off\\",
        \\"@typescript-eslint/no-unused-vars\\": [
          \\"error\\",
          {
            \\"argsIgnorePattern\\": \\"^_\\",
            \\"varsIgnorePattern\\": \\"^(_|global)\\"
          }
        ],
        \\"@typescript-eslint/object-curly-spacing\\": [
          \\"error\\",
          \\"always\\"
        ],
        \\"@typescript-eslint/prefer-readonly-parameter-types\\": \\"off\\",
        \\"@typescript-eslint/quotes\\": [
          \\"error\\",
          \\"single\\"
        ],
        \\"@typescript-eslint/semi\\": [
          \\"error\\",
          \\"never\\"
        ],
        \\"capitalized-comments\\": \\"off\\",
        \\"comma-dangle\\": [
          \\"error\\",
          \\"always-multiline\\"
        ],
        \\"curly\\": [
          \\"error\\",
          \\"multi\\"
        ],
        \\"eslint-comments/no-use\\": \\"off\\",
        \\"etc/no-const-enum\\": \\"off\\",
        \\"etc/no-enum\\": \\"off\\",
        \\"etc/prefer-interface\\": \\"off\\",
        \\"ext/lines-between-object-properties\\": \\"off\\",
        \\"func-names\\": [
          \\"error\\",
          \\"always\\"
        ],
        \\"id-length\\": [
          \\"error\\",
          {
            \\"exceptions\\": [
              \\"t\\"
            ],
            \\"min\\": 2
          }
        ],
        \\"import/extensions\\": \\"off\\",
        \\"import/newline-after-import\\": \\"off\\",
        \\"import/order\\": \\"off\\",
        \\"import/prefer-default-export\\": \\"off\\",
        \\"indent\\": \\"off\\",
        \\"line-comment-position\\": \\"off\\",
        \\"lines-around-comment\\": \\"off\\",
        \\"lines-between-class-members\\": \\"off\\",
        \\"logical-assignment-operators\\": \\"off\\",
        \\"max-len\\": \\"off\\",
        \\"max-statements-per-line\\": \\"off\\",
        \\"no-console\\": \\"error\\",
        \\"no-inline-comments\\": \\"off\\",
        \\"object-curly-spacing\\": [
          \\"error\\",
          \\"always\\"
        ],
        \\"padding-line-between-statements\\": \\"off\\",
        \\"prettier/prettier\\": \\"off\\",
        \\"quote-props\\": [
          \\"error\\",
          \\"consistent-as-needed\\"
        ],
        \\"quotes\\": \\"off\\",
        \\"require-atomic-updates\\": [
          \\"error\\",
          {
            \\"allowProperties\\": true
          }
        ],
        \\"semi\\": [
          \\"error\\",
          \\"never\\"
        ],
        \\"space-before-function-paren\\": [
          \\"error\\",
          \\"always\\"
        ],
        \\"total-functions/no-partial-division\\": \\"off\\",
        \\"unicorn/filename-case\\": \\"off\\",
        \\"unicorn/no-array-for-each\\": \\"off\\",
        \\"unicorn/no-process-exit\\": \\"off\\",
        \\"unicorn/prefer-module\\": \\"off\\",
        \\"unicorn/prefer-node-protocol\\": \\"off\\",
        \\"unicorn/prefer-string-replace-all\\": \\"off\\",
        \\"unicorn/prefer-switch\\": \\"off\\",
        \\"unicorn/prevent-abbreviations\\": [
          \\"error\\",
          {
            \\"allowList\\": {
              \\"args\\": true,
              \\"pkg\\": true,
              \\"str\\": true
            }
          }
        ]
      }
    }
  ],
  \\"rules\\": {}
}",
  "fileName": "",
  "originalFileContent": "",
  "passed": [
    "has-eslint-recommended-rules-extend",
    "has-no-promise-plugin-require-eslint-7",
  ],
  "warnings": [
    "could-have-hardcore-rules-extend-hardcore",
    "could-have-unicorn-rules-extend-plugin-unicorn-all",
  ],
}
`;
